### DISCUSSION TITLE
[RFC] 034 - Layout 实现重构

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#1700] [RFCs]
## 背景

- #1531 
- #772

很多用户反馈说 LobeChat 的反应有些慢，尤其是在页面切换的时候：

> 在使用过程中，确实有时候比较慢，比如从发现tab切换回会话要等一会，可能确实是哪里有渲染影响性能的地方。

> 界面切换慢应该是网络问题，本地部署的话响应速度是OK的，不过能优化就最好了。或者给一个加载动画，显示目前正在加载

最近在重新阅读 NextJS 的文档时，看到有一个关键描述：

> Navigating across multiple root layouts will cause a full page load (as opposed to a client-side navigation). For example, navigating from /cart that uses app/(shop)/layout.js to /blog that uses app/(marketing)/layout.js will cause a full page load. This only applies to multiple root layouts.
> 跨多个根布局导航将导致完整页面加载（与客户端导航相反）。例如，从使用 `app/(shop)/layout.js` 的 `/cart` 导航到使用 `app/(marketing)/layout.js` 的 `/blog` 将导致整个页面加载。这仅适用于多个根布局。
> https://nextjs.org/docs/app/api-reference/file-conventions/layout#root-layouts

所以我大概了解为什么在切换路由的时候会慢了。就是由于目前我们的实现使用了多个根布局导致的。

这个 RFC 针对性进行一次重构，期望可以完全解决切换慢的问题

## 进展

- https://github.com/lobehub/lobe-chat/pull/1736

**Comment by BrandonStudio** [comment]
我有一个问题，这个PWA是完全在本地运行还是怎么？为什么切换路由的时候会卡？
一般如果本地运行的话切换应该不需要时间，还是说切换后的界面中的元素需要重新向服务器发送请求？

**Reply by MapleEve** [reply]
PWA目前只是一个收藏夹+桌面快捷方式

**Reply by BrandonStudio** [reply]
这个next可以编译成wasm吗？

**Reply by arvinxx** [reply]
@BrandonStudio 切换路由会卡的原因我更新说明了。
